src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+  const values = Array.from(new Set(items.map((i) => i[key])));
```

Comment:
Don't make your solution as complicated as possible)
It's redundant here
You need only one array iteration to resolve this task